### PR TITLE
Fix minor annoyances and provide helpful output on test run

### PR DESCRIPTION
**Pull request by skshetry**
## Description

This fixes minor annoyances and provides helpful output on test run.
1. Provide helpful output instead of traceback if user already exists.
2. Show error message if error occurs on `remote_occ`.
3. Test that server is up and running before running tests.
4. Test that testing app is enabled before running tests.
5. Provide message if `.htaccess` cannot be updated.
6. Fix some linter issues from `shellcheck`.

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- locally and on "robot:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/run.sh [hunk 1]
```diff
@@ -930,7 +972,7 @@ for i in "${!BEHAT_SUITES[@]}"
 	do
 	BEHAT_SUITE_OPTION="--suite=${BEHAT_SUITES[$i]}"
 	SUITE_FEATURE_TEXT="${BEHAT_SUITES[$i]}"
-	for rerun_number in $(seq 1 $BEHAT_RERUN_TIMES)
+	for rerun_number in $(seq 1 ${BEHAT_RERUN_TIMES})
 		do
 		if (($BEHAT_RERUN_TIMES > 1))
```
Comment (phil-davis) [inline tests/acceptance/run.sh]:
```suggestion
		if ((${BEHAT_RERUN_TIMES} > 1))
```

This one could also get curly brackets?

File: tests/acceptance/run.sh [hunk 2]
```diff
@@ -560,7 +602,7 @@ else
 		# the remaining number of suites that need to be distributed (could be zero)
 		REMAINING_SUITES=$((${COUNT_ALL_SUITES} - (${DIVIDE_INTO_NUM_PARTS} * ${MIN_SUITES_PER_RUN})))
 
-		if [ ${RUN_PART} -le ${REMAINING_SUITES} ]
+		if [[ ${RUN_PART} -le $REMAINING_SUITES ]]
```
Comment (phil-davis) [inline tests/acceptance/run.sh]: Why do we lose the curly brackets around `$REMAINING_SUITES`

File: tests/acceptance/run.sh [hunk 3]
```diff
@@ -163,7 +163,7 @@ export LANG=C
 # @return occ return code given in the xml data
 function remote_occ() {
 	COMMAND=`echo $3 | xargs`
-	CURL_OCC_RESULT=`curl -k -s -u $1 $2 -d "command=${COMMAND}"`
+	CURL_OCC_RESULT=`curl -k -sS -u $1 $2 -d "command=${COMMAND}"`
```
Comment (phil-davis) [inline tests/acceptance/run.sh]: This looks like magic. What is capital `S` good for?

File: tests/acceptance/run.sh (hunk 2 above)
Comment (skshetry) [reply tests/acceptance/run.sh]: i'll fix that.

File: tests/acceptance/run.sh (hunk 1 above)
Comment (skshetry) [reply tests/acceptance/run.sh]: `shellcheck` did not suggest here. I'll check what to do here.

File: tests/acceptance/run.sh (hunk 3 above)
Comment (skshetry) [reply tests/acceptance/run.sh]: It shows you error message if there is an error even if it's in silent mode.

Comment (skshetry) [reply tests/acceptance/run.sh]: I'll remove this as we dont print the result.

File: tests/acceptance/run.sh [hunk 4]
```diff
@@ -462,24 +494,34 @@ then
...
 	remote_occ ${ADMIN_AUTH} ${OCC_URL} "config:system:set htaccess.RewriteBase --value /${WEBSERVER_PATH}/"
 	remote_occ ${ADMIN_AUTH} ${OCC_URL} "maintenance:update:htaccess"
+	[[ $? -eq 0 ]] || { echo "${HTACCESS_UPDATE_FAILURE_MSG}"; }
 
 	if [ -n "${TEST_SERVER_FED_URL}" ]
 	then
 		WEBSERVER_PATH=$(get_path_from_url ${TEST_SERVER_FED_URL})
 		remote_occ ${ADMIN_AUTH} ${OCC_FED_URL} "config:system:set htaccess.RewriteBase --value /${WEBSERVER_PATH}/"
 		remote_occ ${ADMIN_AUTH} ${OCC_FED_URL} "maintenance:update:htaccess"
+		[[ $? -eq 0 ]] || { echo "${HTACCESS_UPDATE_FAILURE_MSG/local/federated}"; }
```
Comment (phil-davis) [inline tests/acceptance/run.sh]: Note to others: this is a bit of useful bash syntax. The first occurrence of `local` in the `$HTACCESS_UPDATE_FAILURE_MSG` string will be replaced by `federated`

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@skshetry please backport

**Comment by skshetry** [comment]
Backport for `stable10` on #35402

**Pull request merged**